tools/perf/ucc_pt_benchmark.cc

```diff
@@ -0,0 +1,119 @@
+#include <iomanip>
+#include "ucc_pt_benchmark.h"
+#include "core/ucc_mc.h"
+#include "ucc_perftest.h"
+#include "utils/ucc_coll_utils.h"
+
+ucc_pt_benchmark::ucc_pt_benchmark(ucc_pt_benchmark_config cfg,
+                                   ucc_pt_comm *communcator):
+    config(cfg),
+    comm(communcator)
+{
+    coll = new ucc_pt_coll_allreduce(cfg.dt, cfg.mt, cfg.op, cfg.inplace);
```

Comment:
How will you make test with multiple collectives? Will create multiple ucc_pt_benchmark objects in main? Right now it is kind of all tight to a 1 benchmark. You have 1 arg parse, which creates 1 config, with 1 coll. Looks like it must be changed somehow. OR, is the use case is 1 coll_type at a time ? It also fine imho, just making sure i understand.